charts/linkerd2/templates/heartbeat.yaml

```diff
@@ -48,8 +48,8 @@ spec:
             - "-log-level={{.Values.global.controllerLogLevel}}"
             {{- if .Values.global.prometheusUrl }}
             - "-prometheus-url={{.Values.global.prometheusUrl}}"
-            {{- else if .Values.prometheus.enabled }}
-            - "-prometheus-url=http://linkerd-prometheus.{{.Values.global.namespace}}.svc.{{.Values.global.clusterDomain}}:9090"
+            {{- else }}
+            - "-prometheus-url=http://linkerd-prometheus.{{.Values.linkerdVizNamespace}}.svc.{{.Values.global.clusterDomain}}:9090"
```

Comment:
In a future PR, let's change this to discover the linkerd-viz namespace dynamically in the heartbeat code by using the linkerd.io/extension label instead of passing it in as a flag.  For now, let's just hardcode the namespace.